### PR TITLE
Add cascading config finder

### PR DESCRIPTION
**Pull request by isbm**
Boxed plugin can only get a static hard-coded configuration, delivered as a resource inside a Python package. It cannot be overridden without explicit change of a package-delivered data. This is also makes it impossible to programmatically integrate it with Kiwi as a library.

Under these circumstances, a cascading mechanism is required, which is searching for the configuration file in the following order:
1. Environment variable "KIWI_BOXED_PLUGIN_CFG" (full path, the name can be anything)
2. Current directory, from where a software was called
3. `$HOME/.config/kiwi/kiwi_boxed_plugin.yml`
4. `/etc/kiwi_boxed_plugin.yml`
5. Resource (default config, coming with the package)

This PR shares https://github.com/Elektrobit/kiwi-boxed-plugin/pull/1 and is remake of https://github.com/OSInside/kiwi-boxed-plugin/pull/58

### CODE REVIEW

**Comment by isbm** [comment]
@schaefi seems like it successfully failed all the unit tests 😆 Not sure what is this return code, but doesn't look like is related.

**Comment by schaefi** [comment]
> @schaefi seems like it unsuccessfully passed all the unit tests 😆 Not sure what is this return code, but doesn't look like is related.

F401 module imported but not used. You are importing modules in your test code that are not used. See the check target output in the actions log

**Comment by schaefi** [comment]
Screenshot_20231024_221242_Chrome

**Comment by isbm** [comment]
@schaefi should be good now wrt tests. Some shellcheck errors from tox (locally), but I am not sure if they are even related.

File: test/unit/defaults_test.py [hunk 1]
```diff
@@ -1,10 +1,52 @@
...
+    @patch("os.path.exists", lambda f: True)
+    @patch.dict(os.environ, KIWI_BOXED_PLUGIN_CFG="aarchderwelt.conf")
+    def test_get_plugin_config_file_env(self):
+        assert Defaults.get_plugin_config_file() == "aarchderwelt.conf", \
+            "aarch64 aonfiguration from the environment variable do not match"
+
+    @patch("os.path.abspath", lambda f: "/highway/to/hell.conf")
+    @patch("os.path.exists", lambda f: True)
+    def test_get_plugin_config_file_currdir(self):
+        assert Defaults.get_plugin_config_file() == "/highway/to/hell.conf", \
```
Comment (schaefi) [inline test/unit/defaults_test.py:41]: love it :smile:

Comment (isbm) [reply test/unit/defaults_test.py:41]: I mean... stodgy tests should be also fun. 😉

**Pull request merged**